common.cpp

```diff
@@ -85,3 +85,17 @@ void bailout(
 ) {
    bailout(condition, message, "", 0);
 }
+
+/*! Helper function for error handling. err_type default to 0.*/
+[[ noreturn ]] void abort_mpi(const std::string str, const int err_type) {
+   int myRank;
+   MPI_Comm_rank(MPI_COMM_WORLD, &myRank);
+   if (myRank == MASTER_RANK) {
+      if (err_type == 0)
+         std::cerr << str << std::endl;
+      else
+         std::cerr << __FILE__ << ":" << __LINE__ << ": " << str << std::endl;
```

Comment:
This will always print common.cpp:97 and I don't like that.

Reply:
Let's leave it like this for now, and implement https://en.cppreference.com/w/cpp/utility/source_location in a separate PR (which we can then test on all our relevant platforms, since it is C++20 and might not yet be fully available everywhere)